harness/determined/common/schemas/expconf/_gen.py

```diff
@@ -2228,6 +2228,214 @@
+                "integer",
+                "null"
+            ],
+            "minimum": 1,
+            "default": 5
+        },
+        "max_length": {
+            "type": [
+                "object",
+                "null"
+            ],
+            "default": null,
+            "optionalRef": "http://determined.ai/schemas/expconf/v0/check-positive-length.json"
+        },
+        "metric": {
+            "type": [
+                "string",
+                "null"
+            ],
+            "default": null
+        },
+        "smaller_is_better": {
+            "type": [
+                "boolean",
+                "null"
+            ],
+            "default": true
+        },
+        "source_trial_id": {
+            "type": [
+                "integer",
+                "null"
+            ],
+            "default": null
+        },
+        "source_checkpoint_uuid": {
+            "type": [
+                "string",
+                "null"
+            ],
+            "default": null
+        }
+    }
+}
+
+"""
+    ),
+    "http://determined.ai/schemas/expconf/v0/searcher-adaptive.json": json.loads(
+        r"""
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$comment": "this is a historical searcher, not to be used in new experiments",
+    "$id": "http://determined.ai/schemas/expconf/v0/searcher-adaptive.json",
+    "title": "AdaptiveConfig",
+    "type": "object",
+    "additionalProperties": false,
+    "required": [
+        "name"
+    ],
+    "eventuallyRequired": [
+        "budget",
+        "max_length",
+        "metric"
+    ],
+    "properties": {
+        "name": {
+            "const": "adaptive"
+        },
+        "budget": {
+            "type": [
+                "object",
+                "null"
+            ],
+            "default": null,
+            "optionalRef": "http://determined.ai/schemas/expconf/v0/length.json"
+        },
+        "bracket_rungs": {
+            "type": [
+                "array",
+                "null"
+            ],
+            "default": [],
+            "items": {
+                "type": "integer"
+            }
+        },
+        "mode": {
+            "enum": [
+                null,
+                "aggressive",
+                "standard",
+                "conservative"
+            ],
+            "default": "standard"
+        },
+        "divisor": {
+            "type": [
+                "number",
+                "null"
+            ],
+            "exclusiveMinimum": 1,
+            "default": 4
+        },
+        "max_rungs": {
+            "type": [
+                "integer",
+                "null"
+            ],
+            "minimum": 1,
+            "default": 5
+        },
+        "max_length": {
+            "type": [
+                "object",
+                "null"
+            ],
+            "default": null,
+            "optionalRef": "http://determined.ai/schemas/expconf/v0/check-positive-length.json"
```

Comment:
What is the actual difference in the constraints between max_rungs and max_length?

Reply:
I think `max_rungs` affects the shape of the tournament trees that are created and `max_length` is the total training length of the longest-running trial

Reply:
Oh nevermind - I looked at the check-positive-length.json, but missed ITS reference to length.json.